pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  renpy,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "katawa-shoujo-re-engineered";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    /*
+      # alternative source:
+      fetchFromGitea {
+        domain = "codeberg.org";
+        owner = "fhs";
+        repo = "katawa-shoujo-re-engineered";
+    */
+    owner = "fleetingheart";
+    repo = "ksre";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-RYJM/wGVWqIRZzHLUtUZ5mKUrUftDVaOwS1f/EpW6Tk=";
+  };
+
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    echo "#!/${stdenvNoCC.shell}
```

Comment:
I wonder if there is any preference between this and `makeWrapper ... --add-flags ${src} --add-flags run`.

Reply:
Let us wonder together
I have no idea, but to me the makeWrapper version looks cleaner.

There is however something that lets me write the executable as a nix string and pass it as file.
However, other than a few language specific examples, I couldn't find any documentations on it and I didn't want to dig through the stdenv.mkDerivation code.

Reply:
i changed it to use makeWrapper and also added an icon and desktop file